src/main/java/io/gingersnapproject/k8s/CacheRuleInformer.java

```diff
@@ -0,0 +1,220 @@
+package io.gingersnapproject.k8s;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.Map.Entry;
+import java.util.Optional;
+
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import io.gingersnapproject.cdc.DynamicRuleManagement;
+import io.gingersnapproject.cdc.configuration.Configuration;
+import io.gingersnapproject.cdc.configuration.Connector;
+import io.gingersnapproject.cdc.configuration.Rule;
+import io.gingersnapproject.k8s.configuration.KubernetesConfiguration;
+import io.gingersnapproject.proto.api.config.v1alpha1.EagerCacheRuleSpec;
+import io.gingersnapproject.proto.api.config.v1alpha1.KeyFormat;
+import io.gingersnapproject.proto.api.config.v1alpha1.EagerCacheRuleSpec.Builder;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.util.JsonFormat;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.informers.ResourceEventHandler;
+import io.fabric8.kubernetes.client.informers.SharedIndexInformer;
+import io.quarkus.runtime.ShutdownEvent;
+import io.quarkus.runtime.StartupEvent;
+
+@Singleton
```

Comment:
we don't want this to be proxy with ApplicationScope ?

Reply:
yeah seems that ApplicationScope must be preferred over Singleton in Quarkus.
Should I uniform all the occurrences?
```
$ grep -Rn "@Singleton"
src/main/java/io/gingersnapproject/cdc/cache/hotrod/HotRodService.java:36:@Singleton
src/main/java/io/gingersnapproject/k8s/KubernetesClientProducer.java:16:@Singleton
src/main/java/io/gingersnapproject/k8s/CacheRuleInformer.java:37:@Singleton
```

Reply:
ApplicationScoped inject a proxy generated by the framework instead of a plain reference.
Not 100% clear to me too, but I've been convinced by the last line in [this par](https://quarkus.io/guides/cdi#applicationscoped-and-singleton-look-very-similar-which-one-should-i-choose-for-my-quarkus-application)

Reply:
Thanks for the link. I quickly checked the code and I don't think there's any reason we need the `@Singleton` semantics.